host/libhackrf/src/hackrf.c

```diff
@@ -754,6 +759,36 @@ int ADDCALL hackrf_open_by_serial(
 	return hackrf_open_setup(usb_device, device);
 }
 
+int ADDCALL hackrf_open_by_fd(
+	int fd,
```

Comment:
I think we should take `intptr_t sys_dev` here instead and pass it straight through to libusb. Looking back at at the PR for the libusb function (https://github.com/libusb/libusb/pull/242#issuecomment-444600509), they made that choice to make it more general across different platforms so I think we should do the same and leave it up to the user to do any platform-specific conversion/checking.